azext_iot/tests/iothub/configurations/test_iot_config_unit.py

```diff
@@ -899,3 +906,46 @@ def test_config_apply_edge_error(
                 hub_name=mock_target["entity"],
                 content=sample_config_edge_malformed,
             )
+
+
+class TestConfigRead:
+    @pytest.fixture(params=[200])
+    def serviceclient(self, mocker, fixture_ghcs, fixture_sas, request, sample_config_read):
+        service_client = mocker.patch(path_service_client)
+        service_client.side_effect = [
+            build_mock_response(mocker, payload=[{"moduleId" : "$edgeAgent"}, {"moduleId" : "AzureBlobStorageonIoTEdge"}]),
+            build_mock_response(mocker, payload=sample_config_read["moduleTwins"][0]),
+            build_mock_response(mocker, payload=sample_config_read["moduleTwins"][1]),
+        ]
+        return service_client
+
+    @pytest.mark.parametrize(
+        "device_id, hub_name", [("test-device-01", mock_target["entity"])]
+    )
+    def test_config_read_edge(
+        self, fixture_cmd, serviceclient, device_id, hub_name, sample_config_read
+    ):
+        result = subject.iot_edge_read_modules(
+            cmd=fixture_cmd,
+            device_id=device_id,
+            hub_name=mock_target["entity"]
+        )
+
+        assert result == sample_config_read["configuration"]
+
+    @pytest.mark.parametrize(
+        "device_id, hub_name", [("test-device-01", mock_target["entity"])]
+    )
+    def test_config_read_edge_error(
+        self,
+        fixture_cmd,
+        serviceclient_generic_error,
+        device_id,
+        hub_name
+    ):
+        with pytest.raises(CLIError):
```

Comment:
we should get into the habit of checking the error message and specific type of the error raised here...

Reply:
error checking updated, please let me know if this is what you meant, thanks!

Reply:
looks better now